src/main/scala/dspjunctions/JtagAxi.scala

```diff
@@ -0,0 +1,126 @@
+// See LICENSE for license details.
+
+package dspjunctions
+
+import cde._
+import chisel3._
+import chisel3.util._
+import jtag._
+import junctions._
+
+case object DspChainIncludeJtag extends Field[Bool]
+
+class DspJtagIO extends JtagIO {
+  // val TRST = Input(Bool())
+}
+
+trait HasDspJtagParameters {
+  implicit val p: Parameters
+  val includeJtag = try {
+    if (p(DspChainIncludeJtag)) {
+      true
+    } else {
+      false
+    }
+  } catch {
+    case _: ParameterUndefinedException => false
+  }
+}
+
+trait HasDspJtagIO extends HasDspJtagParameters {
+  val p: Parameters
+  val jtag = if (includeJtag) {
+      Some(new DspJtagIO)
+  } else {
+    None
+  }
+}
+
+trait HasDspJtagModule extends HasDspJtagParameters {
+  val io: HasDspJtagIO
+
+  val jtagMaster = if (includeJtag) {
+    Some(Module(new JtagAxiMaster()))
+  } else {
+    None
+  }
+
+  def jtagCtrlAxiMasters: Seq[NastiIO] = jtagMaster.map(m => Seq(m.io.ctrlAxi)).getOrElse(Seq())
+  def jtagDataAxiMasters: Seq[NastiIO] = jtagMaster.map(m => Seq(m.io.ctrlAxi)).getOrElse(Seq())
+
+  def jtagConnect: Unit = (io.jtag, jtagMaster) match {
+    case (Some(jtag), Some(master)) =>
+      jtag <> master.io.jtag
+    case (None, None) =>
+    case _ => throw dsptools.DspException("One of jtagMaster and io.jtag existed, but the other did not!")
+  }
+}
+
+class JtagAxiMaster()(implicit p: Parameters) extends NastiModule()(p) {
+  val io = IO(new Bundle {
+    val jtag    = new DspJtagIO
+    val ctrlAxi = new NastiIO
+    val dataAxi = new NastiIO
+  })
+
+  val tap = Module(new JtagTwoAxiMasterClocked(io.jtag.TCK.asClock, false.B)) // io.jtag.TRST))
```

Comment:
Yes. But TRST is an asynchronous reset, which isn't currently supported. The JTAG block reset also doesn't reset the JTAG block state machine (because if the state machine is in reset, the outer block would be held in reset, and it would never exit reset).